### PR TITLE
Build CI Actions

### PR DESCRIPTION
**Pull request by smasoka**
Actions for build CI 
Checking python
Installing poetry 
poetry install 
poetry run pytest

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: build CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (sjperkins) [inline .github/workflows/main.yml:7]: Lets change this to

```yaml
on: [push, pull_request]
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Configure poetry
+        shell: bash
+        run: poetry config virtualenvs.in-project false
+
+      - name: Install Python Dependencies
+        run: poetry install
+
+      - name: Run complete test suite
+        run: poetry run pytest
```
Comment (sjperkins) [inline .github/workflows/main.yml:47]:
```suggestion
        run: poetry run py.test -s -vvv --flake8 packratt/
```

**Comment by sjperkins** [comment]
@smasoka Can you also change the deploy section to the following:

```yaml
  deploy:
    needs: [test]
    runs-on: ubuntu-latest
    # Run on a push to a tag or master
    if: >
      github.event_name == 'push' &&
      (startsWith(github.event.ref, 'refs/tags') ||
       github.event.ref == 'refs/heads/master')
    steps:
      - name: Set up Python 3.8
        uses: actions/setup-python@v1
        with:
          python-version: 3.8

      - name: Install latest setuptools, wheel, pip
        run: python3 -m pip install -U pip setuptools wheel

      - name: Install dephell
        run: curl -L dephell.org/install | python3

      - name: Checkout source
        uses: actions/checkout@v2
        with:
          fetch-depth: 1

      - name: Convert pyproject.toml to setup.py
        run: dephell deps convert --from=pyproject.toml --to=setup.py

      - name: Build distributions
        run: python setup.py sdist bdist_wheel

      - name: Publish distribution 📦 to Test PyPI
        uses: pypa/gh-action-pypi-publish@master
        with:
          user: __token__
          password: ${{ secrets.test_pypi_token }}
          repository_url: https://test.pypi.org/legacy/
        continue-on-error: true

      - name: Publish distribution 📦 to PyPI
        uses: pypa/gh-action-pypi-publish@master
        if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
        with:
          user: __token__
          password: ${{ secrets.pypi_token }}
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          echo "::set-env name=PATH::$HOME/.poetry/bin:$PATH"
+          source $HOME/.poetry/env
+          poetry --version
+
+      - name: Configure poetry
+        shell: bash
+        run: poetry config virtualenvs.in-project false
+
+      - name: Install Python Dependencies
+        run: poetry install
```
Comment (sjperkins) [inline .github/workflows/main.yml:40]: I think this needs to be

```suggestion
        run: poetry install -E testing
```

Comment (smasoka) [reply .github/workflows/main.yml]: Done.

**Pull request merged**